app/utils/temporal.ts

```diff
@@ -47,12 +55,12 @@ export function getTimestamps(
         // NOTE: We are doing this to avoid issues due to timezone
         myDate.setUTCHours(0, 0, 0, 0);
 
+        // FIXME: Moved this up, but need to consult with @tnagorra
+        timestamps.push(myDate.getTime());
+        increment += 1;
```

Comment:
Moving this up will definitely change the behavior, so we need a test here.

Reply:
Yes, I've added a TODO to add tests later.

Reply:
Let's add tests before we merge this. This is a must.